honeybee_energy/construction.py

```diff
@@ -29,5 +77,480 @@ def to_idf(self):
+
+        Note that 'unshaded' means that all shade materials in the construction
+        are ignored.
+        """
+        if isinstance(self.materials[0], EnergyWindowMaterialSimpleGlazSys):
+            # E+ interprets ~80% of solar heat gain from direct solar transmission
+            return self.materials[0].shgc * 0.8
+        trans = 1
+        for mat in self.materials:
+            if isinstance(mat, _EnergyWindowMaterialGlazingBase):
+                trans *= mat.solar_transmittance
+        return trans
+
+    @property
+    def unshaded_visible_transmittance(self):
+        """The unshaded visible transmittance of the window at normal incidence.
+
+        Note that 'unshaded' means that all shade materials in the construction
+        are ignored.
+        """
+        if isinstance(self.materials[0], EnergyWindowMaterialSimpleGlazSys):
+            return self.materials[0].vt
+        trans = 1
+        for mat in self.materials:
+            if isinstance(mat, _EnergyWindowMaterialGlazingBase):
+                trans *= mat.visible_transmittance
+        return trans
+
+    @property
+    def glazing_count(self):
+        """The number of glazing materials contained within the window construction."""
+        count = 0
+        for mat in self.materials:
+            if isinstance(mat, _EnergyWindowMaterialGlazingBase):
+                count += 1
+        return count
+
+    @property
+    def gap_count(self):
+        """The number of gas gaps contained within the window construction.
+
+        Note that this property will count the distance between shades and glass
+        as a gap in addition to any gas layers.
+        """
+        count = 0
+        for i, mat in enumerate(self.materials):
+            if isinstance(mat, _EnergyWindowMaterialGasBase):
+                count += 1
+            elif isinstance(mat, _EnergyWindowMaterialShadeBase):
+                if i == 0 or count == len(self.materials) - 1:
+                    count += 1
+                else:
+                    count += 2
+        return count
+
+    @property
+    def has_shade(self):
+        """Boolean noting whether there is a shade or blind in the construction."""
+        return self._has_shade
+
+    @property
+    def shade_location(self):
+        """Text noting the location of shade in the construction.
+
+        This will be one of the following: ('Interior', 'Exterior', 'Between', None).
+        None indicates that there is no shade within the construction.
+        """
+        if isinstance(self.materials[0], _EnergyWindowMaterialShadeBase):
+            return 'Exterior'
+        elif isinstance(self.materials[-1], _EnergyWindowMaterialShadeBase):
+            return 'Interior'
+        elif self.has_shade:
+            return 'Between'
+        else:
+            return None
+
+    def temperature_profile(self, outside_temperature=-18, inside_temperature=21):
```

Comment:
This is a very nice method and can be very helpful for visualization too.

Reply:
Glad to hear that you appreciate it!
It was quite a pain in the neck to write correctly for gas layers (which is needed for interior / exterior air films as well as window gaps).  But it has made the u-value calculations of windows much more accurate and I think that you are right that we can produce some good visualizations with it.